containers/src/Kokkos_DynRankView.hpp

```diff
@@ -574,10 +577,98 @@ class DynRankView : private View<DataType*******, Properties...> {
   using view_type::stride_7;            // FIXME: not tested
   using view_type::use_count;
 
-  KOKKOS_FUNCTION reference_type
-  operator()(index_type i0 = 0, index_type i1 = 0, index_type i2 = 0,
-             index_type i3 = 0, index_type i4 = 0, index_type i5 = 0,
-             index_type i6 = 0) const {
+  // Adding shortcut operators for rank-0 to rank-3 for default layouts
+  // and access modalities.
+  // This removes performance overhead for always using rank-7 mapping.
+  // See https://github.com/kokkos/kokkos/issues/7604
+  // When boundschecking is enabled we still go through the underlying
+  // rank-7 View to leverage the error checks there.
+
+  KOKKOS_FUNCTION reference_type operator()() const {
+#ifdef KOKKOS_ENABLE_DEBUG
+    if (rank() != 0u)
+      Kokkos::abort(
+          "DynRankView operator() called with invalid number of arguments.");
```

Comment:
You could also do
```C++
KOKKOS_EXPECTS(rank() != 0u && "DynRankView operator() called with invalid number of arguments");

Reply:
was copied from the `operator[]` do you want me to change it everywhere?

Reply:
Talked with Damien, will change in a follow up everywhere.